packages/vuetify/src/components/VCalendar/util/parser.ts

```diff
@@ -21,7 +21,7 @@ export function getParsedCategories (
       const categoryName = typeof v === 'string' ? v
         : typeof v === 'object' && v && typeof v.categoryName === 'string' ? v.categoryName
           : parsedCategoryText(v, categoryText)
-      return { categoryName }
+      return Object.assign({ categoryName }, v)
```

Comment:
Unless I'm mistaken, `v` here is sometimes a string and sometimes an object?

Reply:
Yes, object.assign will just add each letter as indexed values, but instead of mutating the object depending on the type of v and adding several lines of code, it's just simpler this way. If v is a string, we only need categoryName, otherwise we want all the props of v on the object.

Reply:
I would prefer to be verbose and correct here. You could for example return early if v (bad variable name btw, please use `category` or something similar) is a string, and be left with an object